packages/vaex-core/vaex/column.py

```diff
@@ -382,8 +382,17 @@ def __getitem__(self, slice):
             # arrow and numpy do not like the negative indices, so we set them to 0
             take_indices = indices.copy()
             take_indices[mask] = 0
-        if isinstance(ar_unfiltered, supported_arrow_array_types):
-            ar = ar_unfiltered.take(vaex.array_types.to_arrow(take_indices))
+        # Don't use .take in arrow anymore
+        # https://issues.apache.org/jira/browse/ARROW-9773
+        # slice is zero-copy
+        if isinstance(ar_unfiltered, pa.Array):
+            ar = pa.concat_arrays(
+                [ar_unfiltered.slice(i, 1) for i in take_indices]
+            )
+        elif isinstance(ar_unfiltered, pa.ChunkedArray):
+            ar = pa.concat_arrays(
+                [ar_unfiltered.slice(i, 1).combine_chunks() for i in take_indices]
+            )
```

Comment:
```suggestion
        if isinstance(ar_unfiltered, pa.Array):
            ar = ar_unfiltered.take(vaex.array_types.to_arrow(take_indices))
        elif isinstance(ar_unfiltered, pa.ChunkedArray):
            # Don't use .take in arrow for chunked arrays 
            # https://issues.apache.org/jira/browse/ARROW-9773
            # slice is zero-copy
            ar = pa.concat_arrays(
                [ar_unfiltered.slice(i, 1).combine_chunks() for i in take_indices]
            )
```
it is only an issue for chunked arrays right?
Maybe the best place to put this is vaex.array_types.take ?
There is already a take function there, and if all places in vaex uses that function, we have a nice central place where we can solve this.

Reply:
@maartenbreddels i do not think this is only an issue for chunked_arrays, I think it's an issue with `take` broadly. I definitely like the idea of moving it down to a centralized place, but I think no matter the array type, you need to replace `take` with `slice`

Reply:
https://issues.apache.org/jira/browse/ARROW-9773 only talks about chunked arrays, and I also think it's only a problem for that (since it's more complex).

Reply:
Okay i'll take your word on this one, you certainly know more here. I'll move this into the vaex `array_types.take`